Passing environmental variable to task definition

ethicalmohit opened the pull request:
This will pass the environmental variable to the "docker run" command.

Docker File Changes:
CMD FAASOS_ENV=$FAASOS_ENV node code/index.js

Docker Run Example: 
ubuntu@panther:~$ docker run -it -e "FAASOS_ENV=dev" api-gateway /bin/sh
/app # echo $FAASOS_ENV
dev

Task Definition Json Snippet:
      "environment": [
        {
          "name": "FAASOS_ENV",
          "value": "dev"
        },
        {
          "name": "Version",
          "value": "1.5"
        }
      ],
+label: docsite_pr

##### SUMMARY
This will help the user to pass the multiple environmental variables to the task definition which is used to execute the docker run command.

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
ECS Task Definition

##### ANSIBLE VERSION
```
ansible 2.5.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/ubuntu/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.12 (default, Nov 20 2017, 18:23:56) [GCC 5.4.0 20160609]
```

ethicalmohit committed:
Passing environmental variable to task definition

This will pass the environmental variable to the "docker run" command.

Docker File Changes:
CMD FAASOS_ENV=$FAASOS_ENV node code/index.js

Docker Run Example: 
ubuntu@panther:~$ docker run -it -e "FAASOS_ENV=dev" api-gateway /bin/sh
/app # echo $FAASOS_ENV
dev

Task Definition Json Snippet:
      "environment": [
        {
          "name": "FAASOS_ENV",
          "value": "dev"
        },
        {
          "name": "Version",
          "value": "1.5"
        }
      ],
+label: docsite_pr

dagwieers commented on lib/ansible/modules/cloud/amazon/ecs_taskdefinition.py:
```diff
@@ -97,6 +97,11 @@
           awslogs-group: ecs
           awslogs-region: us-west-2
+      environment:
+         -  name: "FAASOS_ENV"
```
You don't need to quote those strings. We tend to only use quotes when it's strictly necessary, and use single quotes first (for various reasons).

ethicalmohit replied:
Thanks for letting me know. I have fixed this accordingly. Kindly recheck.

ethicalmohit committed:
Removed the double quotes

Removed the double quotes from the name and the value of the environmental variable as requested.

ethicalmohit reviewed (commented):
Changes has been done accordingly.

ethicalmohit commented:
!needs_rebase

beeankha commented on lib/ansible/modules/cloud/amazon/ecs_taskdefinition.py:
```diff
@@ -97,6 +97,11 @@
+         -  name: FAASOS_ENV
+            value: dev
+         -  name: VERSION
+            value: 1.5
```
Spacing should look like:
```
      environment:
      - name: FAASOS_ENV
        value: dev
      - name: VERSION
        value: 1.5
```
(there is too much/incorrect white space)

ethicalmohit committed:
Removed unnecessary spaces.

ethicalmohit commented:
ready_for_review

s-hertel commented:
It would be nice to actually document these suboptions rather than just adding them to examples.

s-hertel committed:
fix whitespace

acozine commented:
Thanks for your interest in the Ansible documentation @ethicalmohit! I see you've deleted your Ansible fork, so I'll close this PR. If you'd like to rejoin the conversation, you can find us on freenode IRC at #ansible-docs.
